### DISCUSSION TITLE
Help with CSV Import nested shape issues please?

### DISCUSSION THREAD

**Discussion by DawMatt** [jgraph/drawio#2350] [Q&A]
Hi,

I've been trying to use the CSV Import feature of Draw.io but it doesn't seem to cope with large/complex data sets. Specifically with lots of nested shapes the generated diagram is comically large and the effort to manually reshape it undermines the value of generating the diagram in the first place.

Issue #2188 was raised about this in August but has received no interest/acknowledgement so far. Sample [CSV import files](https://github.com/jgraph/drawio/files/7011793/DrawIOCSVDoc-LayoutIssue.zip) to reproduce the issue and sample outputs were included.

Does anyone have suggestions or ideas on how CSV import can be used to generate more usable diagrams from large data sets including nested shapes? The output doesn't need to be perfect but it needs to be more compact than the examples generated so far (see below).

Thanks, Matt

![130010139-d94a8e2e-ea97-4e36-bc7c-6a5740b76cb1 1](https://user-images.githubusercontent.com/476705/138816824-13f2a937-b4fa-4c10-91fc-b655177662bd.png)

**Comment by tempest-1973** [comment]
Hi did you ever reslve this issue i am trying to do a similar thing.  I can create a shape and set it as a container but using the csv import I ca not see how to add child shapes

**Comment by wyse03br** [comment]
Any luck on this? I'm trying to create a CSV diagram using nested shapes (or shapes in containers, whichever is easiest) but to no avail. Tks

**Comment by wyse03br** [comment]
Looking around, I found this [comment](https://answers.seibert.group/t/draw-io-can-i-use-complex-shape-in-csv-import/1078):
 
> Because the CSV to diagram conversion is a script, you are limited to the default shapes in the [draw.io](http://draw.io/) shape libraries. The conversion script can’t use custom shapes or container shapes that contain child shapes.
 You can, however, add images (via a URL) to shape labels as in the example in the CSV insert dialog.
